oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/function/AvgHistogramPercentileFunction.java

```diff
@@ -168,38 +178,64 @@ public void combine(final Metrics metrics) {
-                String key = sortedKeys.get(i);
-                final Long value = dataset.get(key);
-
-                count += value;
-                for (int rankIdx = loopIndex; rankIdx < roofs.length; rankIdx++) {
-                    int roof = roofs[rankIdx];
-
-                    if (count >= roof) {
-                        long latency = (i + 1 == sortedKeys.size()) ? Long.MAX_VALUE : Long.parseLong(sortedKeys.get(i + 1));
-                        percentileValues.put(String.valueOf(ranks.get(rankIdx)), latency);
-                        loopIndex++;
-                    } else {
-                        break;
+            final Set<String> keys = summation.keys();
+            for (String key : keys) {
+                long value = 0;
+                if (count.get(key) != 0) {
+                    value = summation.get(key) / count.get(key);
+                    if (value == 0L && summation.get(key) > 0L) {
+                        value = 1;
                     }
                 }
+                dataset.put(key, value);
             }
+            dataset.keys().stream()
+                .map(key -> {
+                    if (key.contains(":")) {
+                        String[] kk = key.split(":");
+                        return Tuple.of(kk[0], key);
+                    } else {
+                        return Tuple.of(DEFAULT_GROUP, key);
+                    }
+                })
+                .collect(groupingBy(Tuple2::_1, mapping(Tuple2::_2, Collector.of(
+                    DataTable::new,
+                    (dt, key) -> dt.put(key.contains(":") ? key.split(":")[1] : key, dataset.get(key)),
+                    DataTable::append))))
+                .forEach((group, subDataset) -> {
+                    long total;
+                    total = subDataset.sumOfValues();
+
+                    int[] roofs = new int[ranks.size()];
+                    for (int i = 0; i < ranks.size(); i++) {
+                        roofs[i] = Math.round(total * ranks.get(i) * 1.0f / 100);
+                    }
+
+                    int count = 0;
+                    final List<String> sortedKeys = subDataset.sortedKeys(Comparator.comparingLong(Long::parseLong));
+
+                    int loopIndex = 0;
+
+                    for (String key : sortedKeys) {
+                        final Long value = subDataset.get(key);
+
+                        count += value;
+                        for (int rankIdx = loopIndex; rankIdx < roofs.length; rankIdx++) {
+                            int roof = roofs[rankIdx];
+
+                            if (count >= roof) {
+                                if (group.equals(DEFAULT_GROUP)) {
+                                    percentileValues.put(String.valueOf(ranks.get(rankIdx)), Long.parseLong(key));
+                                } else {
+                                    percentileValues.put(String.format("%s:%s", group, ranks.get(rankIdx)), Long.parseLong(key));
```

Comment:
So, according to this, the `name:99` is formated from here, right?

Reply:
Exactly. `:` is the splitter.